context-and-compliance/02-testing.md

```diff
@@ -0,0 +1,31 @@
+# 2 - Testing
+
+## 2.1 Test code coverage of 90% minimum
+
+#### Context & justification
+
+Ensuring **Test Driven Development (TDD)** is widely practiced across Made Tech, tests should be written for _every_ piece of code. Test code coverage of 90% minimum **should** indicate that 10% of code is not covered by tests. In some cases, tests cannot be written for every piece of code so 90% is chosen as the minimum. However, all project teams should aim to write tests for 100% of their code.
+
+#### Compliance
+
+* [ ] Test code coverage of 90% minimum
+
+## 2.2 Unit tests that run in under 30 seconds
+
+#### Context & justification
+
+Each feature should be tested in isolation hence unit testing. Unit tests are necessary in order to ensure the code written  is correct. These tests should typically run in under 30 seconds unless there is good reasoning behind why this cannot be achieved.
```

Comment:
The main aim is to accommodate quick feedback loops.

There's some good background + a proposal to require partial testing support: https://github.com/madetech/productionisation/issues/17

Reply:
Changed in https://github.com/madetech/productionisation/pull/57/commits/376095ee6ae4fe3c0edb0b1fe769040789411774.

Would it be a good idea to introduce partial testing support now? This could be revisited in another PR once the checklist item is also updated accordingly. Making a Trello comment regarding this.

Reply:
not for right now. Let's go for the current iteration being tested out first.